common/PersistentCharacter.ts

```diff
@@ -32,24 +32,19 @@ export namespace PersistentCharacter {
   export const GetSearchHint = (character: PersistentCharacter) =>
     character.StatBlock.Type;
 
-  const GetTotalLevelFromString = (levelString: string) => {
+  export const GetTotalLevelFromString = (levelString: string) => {
     const matches = levelString.toString().match(/\d+/g);
     if (!matches) {
-      return "";
+      return 0;
     }
-
     return matches
-      .reduce((total, digitMatch) => {
-        const level = parseInt(digitMatch);
-        if (!isNaN(level)) {
-          return total + level;
-        }
-        return total;
-      }, 0)
-      .toString();
+      .map(n => Number.parseInt(n))
+      .filter(n => !isNaN(n))
+      .reduce((a, b) => a + b, 0);
```

Comment:
Is there some test verifying that this is a refactoring (or more correct than before?)

Reply:
I can add some tests to verify.

The return type of the function does of course change from `string` to `number` -- that was in order to leverage the function for use in the associated `StatReference` for levels (parsing _back_ from string there seemed silly).

The refactoring of the reduce was just a personal preference. I like having functional-style array manips to be divided into several smaller operations when possible, but again just what feels more readable to me. Your call here.

There is **one edge case that changes behavior here**: `GetMetadata` on a level string of containing 0 now returns empty string instead of `"0"`. I didn't really consider the utility of that much... do you expect folks to be using "Level 0" characters, and what should be the behavior there?

Reply:
Level 0 characters should be counted as a level 0 character. This is relevant for the difficulty calculator, since a level 0 character still affects the party size and average party level.